source/common/network/connection_impl.cc

```diff
@@ -94,6 +80,18 @@ ConnectionImpl::ConnectionImpl(Event::DispatcherImpl& dispatcher, int fd,
   file_event_ = dispatcher_.createFileEvent(
       fd_, [this](uint32_t events) -> void { onFileEvent(events); }, Event::FileTriggerType::Edge,
       Event::FileReadyType::Read | Event::FileReadyType::Write);
+
+  if (bind_to_address != nullptr) {
+    int rc = bind_to_address->bind(fd);
+    if (rc < 0) {
+      ENVOY_LOG_MISC(warn, "Bind failure. Failed to bind to {}: {}", bind_to_address->asString(),
```

Comment:
If something goes wrong here, this is going to mega spew the logs in a bad way. I'm guessing internally you have rate limiting on individual log messages. We don't have that in public code. I feel like we should have some kind of stat here for a bind error on connect? Perhaps this could be done like the buffer stats (which would be installed by the time you do the deferred close). At minimum we should probably have some kind of TODO here to think a bit more about how we would want to handle this operationally if we are not going to crash.

Reply:
Maybe a counter and a `debug` level log would make sense.

Reply:
Man, if only we had some utility for logging important error messages infrequently.  Like a macro which logged with exponential back-off with a super cheap (if somewhat lossy) atomic counter to minimize CPU usage and cross thread contention.  And then eventually maybe even tie it into the stats system with the macro taking a custom label or line number for easy monitoring and trace-back of unexpected but not fatal occurrences...  That seems like it'd be super useful for debugging production issues for a large scale proxy.
I'll add porting this feature to my non-critical TODO list :-)

Reply:
FWIW, this is going to be a very infrequently used feature (you are probably the only one who will use it for now), so if you have this covered in some other way and you want to put in a TODO that's fine.